onevizion/EMail.py

```diff
@@ -0,0 +1,186 @@
+			raise ("UserName,Password,and Server are required in the PasswordData json")
+		else:
+			self.server = SMTP['Server']
+			self.userName = SMTP['UserName']
+			self.password = SMTP['Password']
+		if 'Port' in SMTP:
+			self.port = int(SMTP['Port'])
+		if 'TLS' in SMTP:
+			self.tls = SMTP['TLS']
+			self.security = 'STARTTLS'
+		if 'Security' in SMTP:
+			self.security = SMTP['Security']
+		if 'From' in SMTP:
+			self.sender = SMTP['From']
+		else:
+			self.sender = SMTP['UserName']
+		if 'To' in SMTP:
+			if type(SMTP['To']) is list:
+				self.to.extend(SMTP['To'])
+			else:
+				self.to.append(SMTP['To'])
+		if 'CC' in SMTP:
+			if type(SMTP['CC']) is list:
+				self.cc.extend(SMTP['CC'])
+			else:
+				self.cc.append(SMTP['CC'])
+
+
+	def sendmail(self):
+		"""Main work body, sends email with preconfigured attributes
+		"""
+		import mimetypes
+
+		from optparse import OptionParser
+
+		from email import encoders
+		#from email.message import Message
+		from email.mime.audio import MIMEAudio
+		from email.mime.base import MIMEBase
+		from email.mime.image import MIMEImage
+		from email.mime.multipart import MIMEMultipart
+		from email.mime.text import MIMEText
+		msg = MIMEMultipart()
+		msg['To'] = ", ".join(self.to )
+		if self.sender != '':
+			msg['From'] = self.sender
+		else:
+			msg['From'] = self.userName
+		msg['Subject'] = self.subject
+
+		body = self.message + "\n"
+
+		for key,value in self.info.items():
+			body = body + "\n\n" + key + ":"
+			if isinstance(value, basestring):
+				svalue = value.encode('ascii', 'ignore').decode('ascii', 'ignore')
+			else:
+				svalue = str(value)
+			if "\n" in svalue:
+				body = body + "\n" + svalue
+			else:
+				body = body + " " + svalue
+		self.body = body
+
+		part = MIMEText(body, 'plain')
+		msg.attach(part)
+
+		for file in self.files:
+			ctype, encoding = mimetypes.guess_type(file)
+			if ctype is None or encoding is not None:
+				# No guess could be made, or the file is encoded (compressed), so
+				# use a generic bag-of-bits type.
+				ctype = 'application/octet-stream'
+			maintype, subtype = ctype.split('/', 1)
+			if maintype == 'text':
+				fp = open(file)
+				# Note: we should handle calculating the charset
+				attachment = MIMEText(fp.read(), _subtype=subtype)
+				fp.close()
+			elif maintype == 'image':
+				fp = open(file, 'rb')
+				attachment = MIMEImage(fp.read(), _subtype=subtype)
+				fp.close()
+			elif maintype == 'audio':
+				fp = open(file, 'rb')
+				attachment = MIMEAudio(fp.read(), _subtype=subtype)
+				fp.close()
+			else:
+				fp = open(file, 'rb')
+				attachment = MIMEBase(maintype, subtype)
+				attachment.set_payload(fp.read())
+				fp.close()
+				# Encode the payload using Base64
+				encoders.encode_base64(attachment)
+			# Set the filename parameter
+			attachment.add_header('Content-Disposition', 'attachment', filename=file)
+			msg.attach(attachment)
+
+
+
+		before = datetime.utcnow()
+		Message("Sending Email...",1)
```

Comment:
I'm not sure what "Message" is.
Perhaps we need to add ```from onevizion.util import Message``` import statement

Reply:
yes.  I missed that one.